examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,358 @@
+    # @param account the NFT owner address
+    # @param operator the operator address
+    assert self.paused == False, "The contract has been paused"
+    return self.operatorApprovals[account][operator]
+
+@external
+def safeTransferFrom(sender: address, receiver: address, id: uint256, amount: uint256, bytes: bytes32):
+    # @dev transfer token from one address to another.
+    # @param sender the sending account (current owner)
+    # @param receiver the receiving account
+    # @param id the token id that will be sent
+    # @param amount the amount of tokens for the specified id
+    assert not self.paused, "The contract has been paused"
+    assert receiver != ZERO_ADDRESS, "ERC1155: transfer to the zero address"
+    assert receiver != sender, "ERC1155: transfer to self"
+    assert (sender == msg.sender) or (self.operatorApprovals[sender][msg.sender] == True), "Caller is neither owner nor approved operator for this ID"
+    assert self.balances[id][sender] > 0 , "caller does not own this ID or ZERO balance"
+    operator: address = msg.sender
+    fromBalance: uint256 = self.balances[id][sender]
+    assert fromBalance >= amount, "ERC1155: insufficient balance for transfer"
+    self.balances[id][sender] = fromBalance - amount
+    self.balances[id][receiver] += amount
+    log TransferSingle(operator, sender, receiver, id, amount)
+
+@external
+def safeBatchTransferFrom(sender: address, receiver: address, ids: DynArray[uint256, BATCH_SIZE], amounts: DynArray[uint256, BATCH_SIZE], _bytes: bytes32):
+    # @dev transfer tokens from one address to another.
+    # @param sender the sending account
+    # @param receiver the receiving account
+    # @param ids a dynamic array of the token ids that will be sent
+    # @param amounts a dynamic array of the amounts for the specified list of ids.
+    assert not self.paused, "The contract has been paused"
+    assert receiver != ZERO_ADDRESS, "ERC1155: transfer to the zero address"
+    assert (sender == msg.sender) or (self.operatorApprovals[sender][msg.sender] == True), "Caller is neither owner nor approved operator for this ID"
+    assert len(ids) == len(amounts), "ERC1155: ids and amounts length mismatch"
+    assert sender != receiver, "sending to self"
+    operator: address = msg.sender
+    for i in range(BATCH_SIZE):
+        if i >= len(ids):
+            break
+        assert self.balances[ids[i]][sender] > 0 , "caller does not own this ID or ZERO balance"
+        id: uint256 = ids[i]
+        amount: uint256 = amounts[i]
+        fromBalance: uint256 = self.balances[id][sender]
+        assert fromBalance >= amount, "ERC1155: transfer amount exceeds balance"
+        # self.balances[id][msg.sender] = fromBalance - amount
+        self.balances[id][sender] = fromBalance - amount
+        self.balances[id][receiver] += amount
+    
+    log TransferBatch(operator, sender, receiver, ids, amounts)
+
+# misc #
```

Comment:
Missing `URI` function

Reply:
My mistake, there is no `URI` function defined in the base spec (only a `URI` event)

The optional metadata extension defines a `uri(uint256)->string` function that is indeed missing from this implementation